src/Controls/src/Core/PlatformConfiguration/WindowsSpecific/Application.cs

```diff
@@ -28,5 +30,13 @@ public static string GetImageDirectory(BindableObject element)
 			config.Element.SetValue(ImageDirectoryProperty, value);
 			return config;
 		}
+
+		static void OnImageDirectoryChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+#if WINDOWS
+			var config = MauiWinUIApplication.Current.Services.GetService<IImageSourceServiceConfiguration>();
+			config?.SetImageDirectory(newValue?.ToString());
+#endif
```

Comment:
This is what I did for a way to add config for the image sources, but I don't like this as it doesn't fix. But, it needs to be dynamic as in the App.xaml sets this, but that is always long after the app has started running. We probably need a better way to do these things. One main issue is that this was on `Application`, but we don't have an application handler as such. In reality, this is a config thing that really only needs to be set in the builder. But for compat reasons, this happens way later. We could say that this is a config option set in the builder, but unfortunately we still need a way to handle the case of using the platform-specifics.

Reply:
This is probably fine. Like you say "this was on Application" so it's sort of special case

The only variation here I can think is if we create a concept of handlers that are less view attached. For example what we have been talking about around Window.  

Application can gain a handler that might even just be attached to "MauiWinUIApplication"

Application inside Forms is an Element and we probably need to have handlers on Elements

Reply:
Probably. If we had a handler for Application, then this will go away as we can respond to this property change from there.

Reply:
Yep, we probably need this for window too. Especially for Windows which actually has a window